File: javascript/lasagna/lasagna.js
/*
 * Name: lasagna.js
 * Purpose: exercism, javascript track, exercise #?
 * by Z Knight, 2021.10.13
 */

/**
 * the expected oven time in minutes
 */
export const EXPECTED_MINUTES_IN_OVEN = 40;

/**
 * The amount of minutes it takes to prepare a single layer.
 */
const PREPARATION_MINUTES_PER_LAYER = 2;

/**
 * Determines the amount of minutes the lasagna still needs to remain in the
 * oven to be properly prepared.
 *
 * @param {number} actualMinutesInOven
 * @returns {number} the number of minutes remaining
 */
export function remainingMinutesInOven(actualMinutesInOven) {
  return EXPECTED_MINUTES_IN_OVEN - actualMinutesInOven;
}

/**
 * Given a number of layers, determines the total preparation time.
 *
 * @param {number} numberOfLayers
 * @returns {number} the total preparation time
 */
export function preparationTimeInMinutes(numberOfLayers) {
  // Preparation takes two minutes per layer
  return numberOfLayers*2;
}

/**
 * Calculates the total working time. That is, the time to prepare all the layers
 * of lasagna, and the time already spent in the oven.
 *
 * @param {number} numberOfLayers
 * @param {number} actualMinutesInOven
 * @returns {number} the total working time
 */
export function totalTimeInMinutes(numberOfLayers, actualMinutesInOven) {
  return preparationTimeInMinutes(numberOfLayers) + actualMinutesInOven;
}

File: python/guidos-gorgeous-lasagna/lasagna.py
#! /usr/bin/env python
"""
    Name:
        lasagna.py
    Purpose:
        exercism, python track, lasagna:
            Program to help you cook a gorgeous lasagna from your favorite cookbook.
    Written by:
        Z Knight, 2021.10.18
"""
EXPECTED_BAKE_TIME = 40 # minutes
PREPARATION_TIME = 2 # minutes per layer

def bake_time_remaining(elapsed_bake_time):
    """
    Calculate the bake time remaining.

    :param elapsed_bake_time: int baking time already elapsed.
    :return: int remaining bake time derived from 'EXPECTED_BAKE_TIME'.

    Function that takes the actual minutes the lasagna has been in the oven as
    an argument and returns how many minutes the lasagna still needs to bake
    based on the `EXPECTED_BAKE_TIME`.
    """
    return EXPECTED_BAKE_TIME - elapsed_bake_time

def preparation_time_in_minutes(number_of_layers):
    """
    Return preparation time based on number of layers in lasagna
    
    :param number_of_layers: int number of layers in the lasagna
    :return: int number of minutes to prepare the layers

    The preparation time is proportonal to the number of layers,
    and can be calculated by this function.
    """
    return PREPARATION_TIME * number_of_layers

def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    """
    Return elapsed cooking time.

    :param number_of_layers: int number of layers in the lasagna
    :param elapsed_bake_time: baking time elapsed so far
    :return: remaining baking time

    This function takes two numbers representing the number of layers & the time already spent 
    baking and calculates the total elapsed minutes spent cooking the lasagna.
    """
    return preparation_time_in_minutes(number_of_layers) + elapsed_bake_time



